include/eve/module/core/regular/logical_and.hpp

```diff
@@ -14,23 +14,14 @@
 namespace eve
 {
   template<typename Options>
-  struct logical_and_t : strict_elementwise_callable<logical_and_t, Options>
+  struct logical_and_t : logical_elementwise_callable<logical_and_t, Options>
   {
-    template<logical_value T, logical_value U>
-    constexpr EVE_FORCEINLINE auto operator()(T a, U b) const noexcept -> decltype(a && b)
-    requires(eve::same_lanes_or_scalar<T, U>)
-    { return EVE_DISPATCH_CALL(a, b); }
-
-    template<logical_value U>
-    constexpr EVE_FORCEINLINE U  operator()(bool a, U b) const noexcept
-    { return EVE_DISPATCH_CALL(a, b); }
-
-    template<logical_value T>
-    constexpr EVE_FORCEINLINE T  operator()(T a, bool b) const noexcept
-    { return EVE_DISPATCH_CALL(a, b); }
-
-    constexpr EVE_FORCEINLINE bool operator()(bool a, bool b) const noexcept
-    { return EVE_DISPATCH_CALL(a, b); }
+    template<typename T, typename U>
+    constexpr EVE_FORCEINLINE common_logical_t<T, U> operator()(T a, U b) const noexcept
+      requires (same_lanes_or_scalar<T, U> && !arithmetic_simd_value<T> && !arithmetic_simd_value<U>)
```

Comment:
This is a very weird requirement. Can't you just require `relaxed_logical_value` on T and U?

Reply:
what comment. 
Like - ok - this is an unacceptable requirement for an interface. We need something reasonable.